src/lib/spsp.js

```diff
@@ -0,0 +1,152 @@
+  return (yield request({
+    method: 'POST',
+    url: receiver,
+    json: true,
+    body: { amount }
+  })).body
+}
+
+const _quote = function * (client, identifier, sourceAmount, destinationAmount) {
+  if (!client) throw new Error('missing client')
+  if (!identifier) throw new Error('missing identifier')
+
+  const queryInfo = yield _query(identifier)
+  const quoteInfo = yield client.quote({
+    destinationAccount: queryInfo.address,
+    destinationAmount: destinationAmount,
+    sourceAmount: sourceAmount
+  })
+
+  return _createPayment(queryInfo, quoteInfo)
+}
+
+const quoteSource = (client, identifier, amount) => {
+  return co(function * () {
+    if (!amount) throw new Error('missing amount')
+    return yield _quote(client, identifier, amount, undefined)
+  })
+}
+
+const quoteDestination = (client, identifier, amount) => {
+  return co(function * () {
+    if (!amount) throw new Error('missing amount')
+    return yield _quote(client, identifier, undefined, amount)
+  })
+}
+
+const sendPayment = (client, payment) => {
+  return co(function * () {
+    if (!client) throw new Error('missing client')
+    if (!payment) throw new Error('missing payment object')
+    if (!payment.destinationAccount) throw new Error('missing payment.destinationAccount')
+    if (!payment.destinationAmount) throw new Error('missing payment.destinationAmount')
+    if (!payment.sourceAmount) throw new Error('missing payment.sourceAmount')
+    if (!payment.connectorAccount) throw new Error('missing payment.connectorAccount')
+    if (!payment.receiver) throw new Error('missing payment.receiver')
+
+    const request = yield _setup(payment.receiver, payment.destinationAmount)
+    const sender = Sender.createSender({ client })
+    const fulfillment = yield sender.payRequest({
+      sourceAmount: String(payment.sourceAmount),
+      connectorAccount: payment.connectorAccount,
+      destinationAmount: String(payment.destinationAmount),
+      destinationAccount: request.address,
+      destinationMemo: {
+        data: request.data,
+        expires_at: request.expires_at
+      },
+      expiresAt: request.expires_at,
+      executionCondition: request.condition
+    })
+
+    return { fulfillment }
+  })
+}
+
+/** Contains SPSP methods */
+class Client {
+  /**
+    * Create an SPSP client.
+    * @param {Object} opts plugin options
+    * @param {Function} opts._plugin (optional) plugin constructor. Defaults to PluginBells
+    */
+  constructor (opts) {
+    this.client = new IlpCore.Client(Object.assign({
+      _plugin: PluginBells
+    }, opts))
+
+    /**
+      * Get payment params via SPSP request, based on source amount
+      * @param {String} identifier webfinger identifier of receiver
+      * @param {String} sourceAmount Amount that you will send
+      * @returns {Promise.<PaymentParams>} Resolves with the parameters that can be passed to sendPayment
+      */
+    this.quoteSource = quoteSource.bind(null, this.client)
```

Comment:
If you need to bind it to this client object it seems like it might make more sense to just make these normal methods on this object, instead of going for that functional style. I think that style only makes sense if we're passing in the plugin directly